### PR TITLE
feat: Checks if question or challenge is in any of learners courses

### PR DESCRIPTION
**Pull request by VladimirBudjen**

### CODE REVIEW

File: SmartTutorTests/Integration/SubmissionTests.cs [hunk 1]
```diff
@@ -135,7 +153,8 @@ public void Accepts_arrange_task_submission_and_produces_correct_evaluation(Arra
                 true
             }
         };
-
+
```
Comment (VladimirBudjen) [inline SmartTutorTests/Integration/SubmissionTests.cs:156]: I will delete this line in next commit. 😄

File: SmartTutorTests/Unit/SubmissionTests.cs [hunk 2]
```diff
@@ -68,22 +99,24 @@ public void Evaluates_answer_submission(QuestionSubmission submission, List<bool
             correctness.ShouldBe(expectedCorrectness);
         }
 
+        private static readonly Submission Submission = new Submission(1, 1, true, DateTime.Now);
```
Comment (Luburic) [inline SmartTutorTests/Unit/SubmissionTests.cs:102]: Remove this. Submission should be an abstract class as it does not make sense as a standalone object (submission of what?). As a side note, we should avoid sharing state between tests, even when it's immutable - if this test suite grows and more tests start to depend on this shared field, what happens when one test changes it?

Why do you need these fields in the test data? If it turns out we don't, don't forget to remove the additional parameter you added to the QuestionSubmission constructor, and the two constructors you added to Submission.

File: SmartTutor/ProgressModel/Submissions/Submission.cs [hunk 3]
```diff
@@ -13,5 +13,25 @@ public void MarkCorrect()
...
             IsCorrect = true;
         }
+
+        protected Submission()
+        {
+        }
+
+        public Submission(int id, int learnerId, bool isCorrect, DateTime timeStamp)
+        {
+            Id = id;
```
Comment (Luburic) [inline SmartTutor/ProgressModel/Submissions/Submission.cs:23]: Look at my comment in the SubmissionTests class, then come back here.

**Review by Luburic — CHANGES_REQUESTED** [review]
A few minor things to address, including the CodeFactor tip.

**Review by Luburic — APPROVED** [review]
Looks good.

**Pull request merged**